tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,689 @@
+
+Name (CLI): `azsdk package update-changelog-content`
+
+Name (MCP): `azsdk_package_update_changelog`
+
+Purpose: Ensure changelog is auto-generated and has an entry for the upcoming release (management-plane) or provide guidance (data-plane) where manual editing is expected. This tool only modifies changelog content and does not change release date or version numbers.
+
+Inputs:
+
+- `--package-path <abs path>` (required)
+
+Execution Steps (Mgmt-Plane):
+
+1. Invoke changelog generation script or tool.
+2. Normalize section ordering / formatting.
+
+Execution Steps (Data-Plane):
+
+1. Return `noop` result advising manual edit and future version bump timing.
+
+Outputs:
+
+```json
+{"result":"succeeded", "message":"Changelog entry added for 2025-10-16", "next_steps": "update version"}
+```
+
+```json
+{"result":"noop", "message":"Data-plane changelog untouched", "next_steps": "Manually edit the changelog, and update version"}
+```
+
+Failure Modes:
+
+- Missing `CHANGELOG.md` → create template (mgmt-plane) or noop (data-plane).
+Note: we should consdier adding recovery support for the generation scenario after milestone 1.
+- Script failure → `failed` with error output.
+
+##### Update Changelog script
+
+When the tool executes the language-specific `update-changelog` script it will pass the following parameters. All paths are absolute.
+
+| Parameter | Description | Type | Required |
+|-----------|-------------|-----:|:--------:|
+| `SdkRepoPath` | Absolute path to the root folder of the local SDK repository. | string | Yes |
+| `PackagePath` | Absolute path to the root folder of the local SDK project (package). | string | Yes |
+
+Note: The script only needs to cover mgmt-plane.
+
+**Example invocations:**
+
+```powershell
+update-changelog.ps1 \
+      -SdkRepoPath C:\dev\repos\azure-sdk-for-net \
+      -PackagePath C:\dev\repos\azure-sdk-for-net\services\newservice
+```
+
+#### 2. Version Update Tool
+
+Name (CLI): `azsdk package update-version`
+
+Name (MCP): `azsdk_package_update_version`
+
+Purpose: Increment and propagate package version across all authoritative version surfaces (e.g., `pom.xml`, `_version.py`, `*.csproj`, `package.json`, `go.mod`, `CHANGELOG.md`).
+
+Inputs:
+
+- `--package-path <abs path>` (required)
+- `--release-type <stable|beta>` (optional)
+- `--version` (optional)
+- `--release-date` (optional)
+
+Execution Steps:
+
+1. Compute a version based on the given information (mgmt-plane).
+2. Apply updates across language-specific files.
+3. Validate consistency (all files now share identical new version).
+4. For data-plane outside release: return `noop`.
+5. For data-plane prepare release: use current version from the metadata file or from user input (support may be post-milestone-1).
+
+> **Note:** In milestone 1, Java management-plane packages will follow the same version update strategy as data-plane packages. Adjustments for Java-specific workflows may be introduced in later versions.
```

Comment:
this should be included in the exclusion section

Reply:
I included this in the exception too.